official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py

```diff
@@ -0,0 +1,309 @@
+      net_down_scale: an `int` that down scales the image width and height to
+        the closest multiple of net_down_scale.
+      max_process_size: an `int` for maximum image width and height.
+      min_process_size: an `int` for minimum image width and height ,
+      max_num_instances: an `int` number of maximum number of instances in an image.
+      random_flip: a `bool` if True, augment training with random horizontal flip.
+      masks: a `Tensor`, `List` or `numpy.ndarrray` for anchor masks.
+      aug_rand_saturation: `bool`, if True, augment training with random
+        saturation.
+      aug_rand_brightness: `bool`, if True, augment training with random
+        brightness.
+      aug_rand_zoom: `bool`, if True, augment training with random
+        zoom.
+      aug_rand_hue: `bool`, if True, augment training with random
+        hue.
+      anchors: a `Tensor`, `List` or `numpy.ndarrray` for bounding box priors.
+      seed: an `int` for the seed used by tf.random
+    """
+    self._net_down_scale = 2**max_level
+
+    self._num_classes = num_classes
+    self._image_w = (image_w // self._net_down_scale) * self._net_down_scale
+    self._image_h = self._image_w if image_h is None else (
+        image_h // self._net_down_scale) * self._net_down_scale
+
+    self._max_process_size = max_process_size
+    self._min_process_size = min_process_size
+    self._fixed_size = fixed_size
+
+    self._anchors = anchors
+    self._masks = {
+        key: tf.convert_to_tensor(value) for key, value in masks.items()
+    }
+    self._use_tie_breaker = use_tie_breaker
+
+    self._jitter_im = 0.0 if jitter_im is None else jitter_im
+    self._jitter_boxes = 0.0 if jitter_boxes is None else jitter_boxes
+    self._max_num_instances = max_num_instances
+    self._random_flip = random_flip
+
+    self._aug_rand_saturation = aug_rand_saturation
+    self._aug_rand_brightness = aug_rand_brightness
+    self._aug_rand_zoom = aug_rand_zoom
+    self._aug_rand_hue = aug_rand_hue
+
+    self._seed = seed
+
+    if dtype == 'float16':
+      self._dtype = tf.float16
+    elif dtype == 'bfloat16':
+      self._dtype = tf.bfloat16
+    elif dtype == 'float32':
+      self._dtype = tf.float32
+    else:
+      raise Exception(
+          'Unsupported datatype used in parser only {float16, bfloat16, or float32}'
+      )
+
+  def _build_grid(self, raw_true, width, batch=False, use_tie_breaker=False):
+    mask = self._masks
+    for key in self._masks.keys():
+      if not batch:
+        mask[key] = preprocessing_ops.build_grided_gt(
+            raw_true, self._masks[key], width // 2**int(key), self._num_classes,
+            raw_true['bbox'].dtype, use_tie_breaker)
+      else:
+        mask[key] = preprocessing_ops.build_batch_grided_gt(
+            raw_true, self._masks[key], width // 2**int(key), self._num_classes,
+            raw_true['bbox'].dtype, use_tie_breaker)
+
+      mask[key] = tf.cast(mask[key], self._dtype)
```

Comment:
You don't want to cast labels to dtype, only the image.

Reply:
Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef